app/src/main/java/com/epfl/drawyourpath/challenge/TrophyViewAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.epfl.drawyourpath.challenge
+
+import android.content.res.AssetManager
+import android.graphics.drawable.Drawable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.epfl.drawyourpath.R
+import java.time.LocalDate
+
+
+/**
+ * used in a recycler view to display the [Trophy]
+ */
+class TrophyViewAdapter(private val trophies: List<Pair<Trophy, LocalDate>>) :
+    RecyclerView.Adapter<TrophyViewAdapter.ViewHolder>() {
+
+    /**
+     * Custom view holder using a custom layout for trophies
+     */
+    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val text: TextView
+        val dateText: TextView
+        val image: ImageView
+        val assets: AssetManager
+

```

Comment:
You can consider the number of time the user win the same trophy(like explain before)(the most of the running app used this concept)